rest-api/dao/participant_dao.py

```diff
@@ -201,6 +202,26 @@ def from_client_json(self, resource_json, id_=None, expected_version=None, clien
         withdrawalStatus=resource_json.get('withdrawalStatus'),
         suspensionStatus=resource_json.get('suspensionStatus'))
 
+  def add_missing_hpo_from_site(self, participant_id, site_id):
```

Comment:
session should be passed into this method... everything should happen in the same transaction.

Also, we already have code that updates participant summaries when physical measurements are updated here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/physical_measurements_dao.py#L252

And fetch the participant summary when inserting a biobank order here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/biobank_order_dao.py#L114

It would be better to avoid looking up this data twice, and instead pass in the participant summary, etc. into a common method.

Reply:
OK. I believe as long as you use the same session, it shouldn't incur extra RPCs to look up the same objects multiple times, so hopefully this is OK.